### ISSUE TITLE
Security - uploading executable files with finder and assets manager

### ISSUE THREAD

**Issue by raffaelj** [agentejo/cockpit#839]
While testing https://github.com/agentejo/cockpit/issues/706#issuecomment-413164374 I discovered an easy way to manipulate files as non-admin by uploading a php file with the finder or the assets manager and calling the file afterwards. This is a big security hole for uses cases, where someone is allowed to upload images etc.

Is there a way to restrict file types for uploads and I just missed the option?

And restricting file types might work for the assets manager when the finder is disabled. Otherwise a file called logo.png could easily be renamed to evilcode.php with the finder.

**Comment by aheinze** [comment]
just landed in next branch: https://github.com/agentejo/cockpit/commit/8fe10aec79df1fa4ac78e934b79b1fedb09b86ea

you can now define allowed file types for finder + assets:

via global setting:

```
allowed_uploads: pdf, png, jpg, zip
```

or for each user group:

```
groups:
    author:
        $admin: false
        $vars:
            finder.path: /storage/upload
            finder.allowed_uploads: pdf, png, jpg
            assets.allowed_uploads: png, jpg
```

**Comment by raffaelj** [comment]
Thanks. I feel much more secure now. I did a few tests. The global setting seems to work. Uploading unlisted files fails.

The group settings didn't seem to work until I realized, that the groups addon overwrites group settings in config.yaml :man_facepalming: 

You test for the file ending and not for the mime type, right?

I was able to upload a php file with the name `evil.png`. Calling it in the browser leads to a mime type failure and calling it through the api `api/cockpit/image` leads to a `...<b>Warning</b>:  Division by zero...`. This is OK.

But there is a mime type detection already. `api/cockpit/assets` tells me:
```json
"path": "/2018/08/18/5b77d177093d0evil.png",
"title": "evil.png",
"mime": "text/x-php",
"description": "",
"tags": [],
"size": 22,
"image": true,
"video": false,
"audio": false,
"archive": false,
"document": false,
"code": false,
"created": 1534579063,
"modified": 1534579063,
"_by": "5b0fb863a1e1cdoc1879762079",
"width": null,
"height": null,
"colors": [],
"_id": "5b77d1773338620b440003b0"
```

And I wasn't able to upload a `.jpeg` file if only `.jpg` is in the allowed list.

Checking against the mime type instead should eliminate the both cases above.

The error massage "File(s) failed to uploaded." should be something like "Didn't upload. Allowed file types are JPEG, PNG, GIF, PDF" so users can interpret the error message.
